### PR TITLE
Bugfix/parameters decoding

### LINKED ISSUE #224: Exception: ValueError: not enough values to unpack (expected 2, got 1)
I am following the guide on: https://github.com/microsoftgraph/msgraph-sdk-python to build a Python package that can pull all my devices from Azure AD (Entra). I have confirmed good working API credentials and that they have the right permissions:

`async def get_AADdevices():
    MScredential = ClientSecretCredential(
        tenant_id=os.environ['MSGraphTenantID'],
        client_id=os.environ['MSGraphClientID'],
        client_secret=os.environ['MSGraphClientSecret']
    )
    scopes = ['https://graph.microsoft.com/.default']
    client = GraphServiceClient(credentials=MScredential, scopes=scopes)
    devices = await client.devices.get()
`

I am getting the following error when it calls the await client.devices.get():

`
Executed 'Functions.MachineReconciliation' (Failed, Id=d6a7b205-25fc-4dfa-ad22-3621ed0b4144, Duration=9ms)
System.Private.CoreLib: Exception while executing function: Functions.MachineReconciliation. System.Private.CoreLib: Result: Failure
Exception: ValueError: not enough values to unpack (expected 2, got 1)
Stack:   File "C:\Program Files\Microsoft\Azure Functions Core Tools\workers\python\3.10/WINDOWS/X64\azure_functions_worker\dispatcher.py", line 479, in _handle__invocation_request
    call_result = await self._loop.run_in_executor(
  File "C:\Users\<user>\AppData\Local\Programs\Python\Python310\lib\concurrent\futures\thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Program Files\Microsoft\Azure Functions Core Tools\workers\python\3.10/WINDOWS/X64\azure_functions_worker\dispatcher.py", line 752, in _run_sync_func
    return ExtensionManager.get_sync_invocation_wrapper(context,
  File "C:\Program Files\Microsoft\Azure Functions Core Tools\workers\python\3.10/WINDOWS/X64\azure_functions_worker\extension.py", line 215, in _raw_invocation_wrapper
    result = function(**args)
  File "C:\Users\<path>\function_app.py", line 17, in MachineReconciliation
    asyncio.run(get_AADdevices())
  File "C:\Users\<user>\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\<user>\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 649, in run_until_complete
    return future.result()
  File "C:\Users\<path>\function_app.py", line 40, in get_AADdevices
    devices = await client.devices.get()
  File "C:\Users\<path>\.venv\lib\site-packages\msgraph\generated\devices\devices_request_builder.py", line 70, in get
    return await self.request_adapter.send_async(request_info, DeviceCollectionResponse, error_mapping)
  File "C:\Users\<path>\.venv\lib\site-packages\kiota_http\httpx_request_adapter.py", line 162, in send_async
    parent_span = self.start_tracing_span(request_info, "send_async")
  File "C:\Users\<path>\.venv\lib\site-packages\kiota_http\httpx_request_adapter.py", line 131, in start_tracing_span
    uri_template = ParametersNameDecodingHandler.decode_uri_encoded_string(
  File "C:\Users\<path>\.venv\lib\site-packages\kiota_http\middleware\parameters_name_decoding_handler.py", line 92, in decode_uri_encoded_string
    name, value = name_value.split('=')
`

### PR DESCRIPTION
**Pull request by samwelkanda**
Fixes #224

### CODE REVIEW

**Comment by baywet** [comment]
@samwelkanda I believe this PR is not needed anymore since #225 was merged. Thoughts?

**Comment by samwelkanda** [comment]
Let me do local tests on those changes and advise @baywet

**Comment by samwelkanda** [comment]
@baywet @ladycarbonfiber  

From my testing

The following tests cases are passing
```py
("https://graph.microsoft.com?%24count=true&query=%24top&created%2din=2022-10-05&q=1%2b2&q2=M%26A&subject%2ename=%7eWelcome&%24empty",
     "https://graph.microsoft.com?$count=true&query=%24top&created-in=2022-10-05&q=1%2b2&q2=M%26A&subject.name=%7eWelcome&$empty"),
    ("http://localhost", "http://localhost"),
    ("https://google.com/?q=1%2b2", "https://google.com/?q=1%2b2"),
    ("https://google.com/?q=M%26A", "https://google.com/?q=M%26A"),
    ("https://google.com/?q=1%2B2", "https://google.com/?q=1%2B2"), # Values are not decoded
     ("https://google.com/?q=M%26A", "https://google.com/?q=M%26A"), # Values are not decoded
```

The following are failing
```py
("http://localhost?%24select=diplayName&api%2Dversion=2", "http://localhost?$select=diplayName&api-version=2"),
("http://localhost?%24select=diplayName&api%7Eversion=2", "http://localhost?$select=diplayName&api~version=2"),
("http://localhost?%24select=diplayName&api%2Eversion=2", "http://localhost?$select=diplayName&api.version=2"),
("http://localhost:888?%24select=diplayName&api%2Dversion=2", "http://localhost:888?$select=diplayName&api-version=2"),
("https://google.com/?q%2D1=M%26A", "https://google.com/?q-1=M%26A"), # Values are not decoded but params are
("https://google.com/?q%2D1&q=M%26A=M%26A", "https://google.com/?q-1&q=M%26A=M%26A"), # Values are not decoded but params are
```

`%2D`, `%7E` `%2E` are not decoded to `-`, '~' and `.` respectively.

For `("https://google.com/?q%2D1&q=M%26A=M%26A", "https://google.com/?q-1&q=M%26A=M%26A"),`

I get the error `ValueError: too many values to unpack (expected 2)`

**Comment by baywet** [comment]
Thanks for the detailed analysis @samwelkanda 
Would you mind rebasing your PR (dropping the changes in the handler, keeping the tests additions), this way we can work on amending the changes submitted by @ladycarbonfiber ?

File: tests/middleware_tests/test_parameters_name_decoding_handler.py [hunk 1]
```diff
@@ -34,14 +34,26 @@ async def test_decodes_query_parameter_names_only():
     """
     Test that only query parameter names are decoded
     """
-    encoded_url = "https://graph.microsoft.com?%24count=true&query=%24top&created%2din=2022-10-05&q=1%2b2&q2=M%26A&subject%2ename=%7eWelcome&%24empty"
-    expected_url = "https://graph.microsoft.com?$count=true&query=%24top&created-in=2022-10-05&q=1%2b2&q2=M%26A&subject.name=%7eWelcome&$empty"
-
+    encoded_decoded = [
+    ("http://localhost?%24select=diplayName&api%2Dversion=2", "http://localhost?$select=diplayName&api-version=2"),
```
Comment (baywet) [inline tests/middleware_tests/test_parameters_name_decoding_handler.py:38]: can we keep a case with a valueless query parameter? ($empty in the previous version)

**Review by baywet — CHANGES_REQUESTED** [review]

**Review by baywet — APPROVED** [review]

**Pull request merged**